src/Database/PostgreSQL/PQTypes/Model/Index.hs

```diff
@@ -24,40 +27,74 @@ import qualified Data.Text.Encoding as T
 
 data TableIndex = TableIndex {
   idxColumns :: [RawSQL ()]
+, idxMethod  :: IndexMethod
 , idxUnique  :: Bool
 , idxWhere   :: Maybe (RawSQL ())
 } deriving (Eq, Ord, Show)
 
+data IndexMethod =
+    BTree
+  | GIN
+  deriving (Eq, Ord)
+
+instance Show IndexMethod where
+    show BTree = "btree"
+    show GIN   = "gin"
+
+instance Read IndexMethod where
+    readsPrec _ "btree" = [(BTree,"")]
```

Comment:
Yeah, I'd make this case-insensitive. You can use view patterns: `readsPrec _ (map toLower -> "btree") = ...`.